src/node.cc

```diff
@@ -3682,9 +3682,10 @@ static void ParseArgs(int* argc,
       force_repl = true;
     } else if (strcmp(arg, "--no-deprecation") == 0) {
       no_deprecation = true;
-    } else if (strcmp(arg, "--napi-modules") == 0) {
-      load_napi_modules = true;
-    } else if (strcmp(arg, "--napi-modules=yes") == 0) {
+    } else if (strcmp(arg, "--napi-modules") == 0 ||
+               strcmp(arg, "--napi-modules=yes") == 0) {
+      fprintf(stderr, "Warning: N-API is an experimental feature "
+        "and could change at any time.\n");
```

Comment:
I don't understand... `process.emitWarning()` is a JavaScript API. From where would it be called?

This was modeled after the [similar warning for the experimental tracing feature](https://github.com/nodejs/node/commit/ba4847e879424ad173289e8fb96cc86a09ee899b#diff-cd53544f44aab2c697bcd7b6a57f23ccR4483).